File: PARTAGER/Structures/PointProjection.cs
namespace FCGP
{
    /// <summary> structure qui sert à véhiculer un point de coordonnées Grille (X,Y ou PointD, Zone et Hémisphère pour un point de coordonnées UTM </summary>
    internal struct PointProjection
    {
        internal double X
        {
            get
            {
                return Coordonnees.X;
            }
            set
            {
                Coordonnees.X = value;
            }
        }
        internal double Y
        {
            get
            {
                return Coordonnees.Y;
            }
            set
            {
                Coordonnees.Y = value;
            }
        }
        internal PointD Coordonnees;
        internal int Zone;
        internal char Hemisphere;
        internal PointProjection(PointD Point, int Z = 0, char H = default)
        {
            Coordonnees = Point;
            Zone = Z;
            Hemisphere = H;
        }
    }
}

File: PARTAGER/Structures/PointProjection.vb
''' <summary> structure qui sert à véhiculer un point de coordonnées Grille (X,Y ou PointD, Zone et Hémisphère pour un point de coordonnées UTM </summary>
Friend Structure PointProjection
    Friend Property X As Double
        Get
            Return Coordonnees.X
        End Get
        Set(value As Double)
            Coordonnees.X = value
        End Set
    End Property
    Friend Property Y As Double
        Get
            Return Coordonnees.Y
        End Get
        Set(value As Double)
            Coordonnees.Y = value
        End Set
    End Property
    Friend Coordonnees As PointD
    Friend Zone As Integer
    Friend Hemisphere As Char
    Friend Sub New(Point As PointD, Optional Z As Integer = 0, Optional H As Char = Nothing)
        Coordonnees = Point
        Zone = Z
        Hemisphere = H
    End Sub
End Structure